<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>735</width>
    <height>612</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <property name="margin">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QFrame" name="widgetFrameOuter">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <property name="class" stdset="0">
       <string>PeripheralCollection</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="spacing">
        <number>5</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>5</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QFrame" name="frame_4">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <property name="spacing">
           <number>3</number>
          </property>
          <property name="margin">
           <number>2</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushButton">
            <property name="minimumSize">
             <size>
              <width>94</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>Start</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_3">
            <property name="minimumSize">
             <size>
              <width>94</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>Stop</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QProgressBar" name="progressBar">
            <property name="styleSheet">
             <string notr="true">QProgressBar:horizontal {
border: 1px solid gray;
border-radius: 3px;
background: white;
padding: 1px;
text-align: right;
margin-right: 5ex;
}
QProgressBar::chunk:horizontal {
background: qlineargradient(x1: 0, y1: 0.5, x2: 1, y2: 0.5, stop: 0 #467, stop:1 white);
margin-right: 1px; /* space */
width: 5px;
}</string>
            </property>
            <property name="maximum">
             <number>2500</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="format">
             <string>%v</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QProgressBar" name="timerProgress">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>18</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Time left before the stopwatch overflows.
 Oscillations must finish within this time limit</string>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="maximum">
             <number>60</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="textVisible">
             <bool>true</bool>
            </property>
            <property name="format">
             <string>%v Seconds to timeout</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="frame_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>50</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <property name="class" stdset="0">
          <string>PeripheralCollectionInner</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="margin">
           <number>2</number>
          </property>
          <item>
           <widget class="QTableWidget" name="resultsTable">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>360</width>
              <height>0</height>
             </size>
            </property>
            <property name="toolTip">
             <string>List of time periods for complete oscillations.
Click and drag to select points.
Hold down CTRL key and click to select non consecutive points</string>
            </property>
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <property name="textElideMode">
             <enum>Qt::ElideMiddle</enum>
            </property>
            <property name="gridStyle">
             <enum>Qt::DashLine</enum>
            </property>
            <property name="rowCount">
             <number>100</number>
            </property>
            <property name="columnCount">
             <number>1</number>
            </property>
            <attribute name="horizontalHeaderDefaultSectionSize">
             <number>150</number>
            </attribute>
            <attribute name="horizontalHeaderMinimumSectionSize">
             <number>40</number>
            </attribute>
            <attribute name="horizontalHeaderStretchLastSection">
             <bool>true</bool>
            </attribute>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <column/>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="10" column="1">
              <widget class="QComboBox" name="comboBox">
               <item>
                <property name="text">
                 <string>Simple Pendulum</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Rod Pendulum</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="8" column="0" colspan="2">
              <widget class="Line" name="line">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
               <property name="lineWidth">
                <number>1</number>
               </property>
               <property name="midLineWidth">
                <number>0</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="13" column="0">
              <widget class="QPushButton" name="pushButton_6">
               <property name="text">
                <string>Calculate</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0" colspan="2">
              <widget class="Line" name="line_2">
               <property name="minimumSize">
                <size>
                 <width>30</width>
                 <height>0</height>
                </size>
               </property>
               <property name="lineWidth">
                <number>1</number>
               </property>
               <property name="midLineWidth">
                <number>0</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="15" column="0">
              <spacer name="verticalSpacer">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="5" column="1">
              <widget class="QLabel" name="averageLabel">
               <property name="text">
                <string>result</string>
               </property>
              </widget>
             </item>
             <item row="10" column="0">
              <widget class="QLabel" name="averageLabel_2">
               <property name="text">
                <string>Value of g using :</string>
               </property>
              </widget>
             </item>
             <item row="11" column="0">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Length (cm)</string>
               </property>
              </widget>
             </item>
             <item row="5" column="0">
              <widget class="QPushButton" name="pushButton_4">
               <property name="toolTip">
                <string>Obtain average of data points selected in the table</string>
               </property>
               <property name="text">
                <string>Average Selected points&gt;</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Data Analysis</string>
               </property>
               <property name="class" stdset="0">
                <string>headingLabel</string>
               </property>
              </widget>
             </item>
             <item row="13" column="1">
              <widget class="QLabel" name="gLabel">
               <property name="text">
                <string>g:</string>
               </property>
              </widget>
             </item>
             <item row="11" column="1">
              <widget class="QDoubleSpinBox" name="lenBox"/>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_3">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Points to acquire</string>
               </property>
              </widget>
             </item>
             <item row="12" column="0" colspan="2">
              <widget class="Line" name="line_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QSpinBox" name="pointBox">
               <property name="minimum">
                <number>5</number>
               </property>
               <property name="maximum">
                <number>1000</number>
               </property>
              </widget>
             </item>
             <item row="1" column="0" colspan="2">
              <widget class="Line" name="line_4">
               <property name="minimumSize">
                <size>
                 <width>30</width>
                 <height>0</height>
                </size>
               </property>
               <property name="lineWidth">
                <number>1</number>
               </property>
               <property name="midLineWidth">
                <number>0</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="2" column="0" colspan="2">
              <widget class="Line" name="line_5">
               <property name="minimumSize">
                <size>
                 <width>30</width>
                 <height>0</height>
                </size>
               </property>
               <property name="lineWidth">
                <number>1</number>
               </property>
               <property name="midLineWidth">
                <number>0</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="16" column="0">
              <widget class="QPushButton" name="pushButton_2">
               <property name="text">
                <string>Clear Table</string>
               </property>
              </widget>
             </item>
             <item row="16" column="1">
              <widget class="QPushButton" name="pushButton_5">
               <property name="text">
                <string>Save Table</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>735</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>run()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>97</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>646</x>
     <y>202</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>194</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>445</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>calcAvg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>568</x>
     <y>194</y>
    </hint>
    <hint type="destinationlabel">
     <x>592</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setPendulumType(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>720</x>
     <y>236</y>
    </hint>
    <hint type="destinationlabel">
     <x>722</x>
     <y>276</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>calculateg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>568</x>
     <y>311</y>
    </hint>
    <hint type="destinationlabel">
     <x>734</x>
     <y>414</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pointBox</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>setTotalPoints()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>652</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>750</x>
     <y>84</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>677</x>
     <y>591</y>
    </hint>
    <hint type="destinationlabel">
     <x>686</x>
     <y>617</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>clearTable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>531</x>
     <y>588</y>
    </hint>
    <hint type="destinationlabel">
     <x>529</x>
     <y>633</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>run()</slot>
  <slot>delete_curve()</slot>
  <slot>stop()</slot>
  <slot>calcAvg()</slot>
  <slot>downloadData()</slot>
  <slot>setPendulumType(int)</slot>
  <slot>calculateg()</slot>
  <slot>setTotalPoints()</slot>
  <slot>saveData()</slot>
  <slot>clearTable()</slot>
 </slots>
</ui>
